websites/management/commands/unpublish_sites.py

```diff
@@ -0,0 +1,125 @@
+                        | Q(type=CONTENT_TYPE_RESOURCE_COLLECTION)
+                    ),
+                    (
+                        Q(metadata__courses__icontains=website.name)
+                        | Q(metadata__resources__content__icontains=website.name)
+                    ),
+                    website__name=settings.ROOT_WEBSITE_NAME,
+                )
+                course_content_dependencies = WebsiteContent.objects.filter(
+                    ~Q(website__name=website.name),
+                    type=CONTENT_TYPE_PAGE,
+                    markdown__icontains=website.name,
+                )
+                course_dependencies = Website.objects.filter(
+                    ~Q(name=website.name), metadata__icontains=website.name
+                )
+                # if there are dependencies, don't unpublish and list dependencies
+                if (
+                    ocw_www_dependencies
+                    or course_dependencies
+                    or course_content_dependencies
+                ):
+                    self.stdout.write(
+                        "Not unpublishing " + website.name + " due to dependencies:"
+                    )
+                    if ocw_www_dependencies:
+                        self.stdout.write(
+                            json.dumps(
+                                {
+                                    "ocw_www": WebsiteContentSerializer(
+                                        instance=ocw_www_dependencies, many=True
+                                    ).data
+                                }
+                            )
+                        )
+                    if course_dependencies:
+                        self.stdout.write(
+                            json.dumps(
+                                {
+                                    "course": WebsiteBasicSerializer(
+                                        instance=course_dependencies, many=True
+                                    ).data
+                                }
+                            )
+                        )
+                    if course_content_dependencies:
+                        self.stdout.write(
+                            json.dumps(
+                                {
+                                    "course_content": WebsiteContentSerializer(
+                                        instance=course_content_dependencies, many=True
+                                    ).data
+                                }
+                            )
+                        )
+                else:  # unpublish
+                    self.stdout.write("Unpublishing " + website.name)
+                    Website.objects.filter(pk=website.pk).update(
+                        unpublish_status=PUBLISH_STATUS_NOT_STARTED,
+                        last_unpublished_by=user_id,
+                    )
+                    trigger_unpublished_removal(website)
```

Comment:
This is going to trigger the `remove-sites-pipeline` for every individual site, which is unnecessary.  
Maybe try this instead:

```python
else:  # unpublish
    self.stdout.write("Unpublishing " + website.name)
    Website.objects.filter(pk=website.pk).update(
        unpublish_status=PUBLISH_STATUS_NOT_STARTED,
        last_unpublished_by=user_id,
      )
      site_pipeline = api.get_site_pipeline(website)
      site_pipeline.pause_pipeline(VERSION_LIVE)
removal_pipeline = api.get_unpublished_removal_pipeline()
removal_pipeline.unpause()
removal_pipeline.trigger()    
```

Reply:
Might also be good to give some indication that something was done at the end, like "`n` sites were unpublished" or "The following sites were unpublished: `[list]`"

Reply:
Yes please

Reply:
Updated in 9b2e6ed52d822fba85c459bcc7e67aeb81be9400.